### DISCUSSION TITLE
Find my configuration for a given rule?

### DISCUSSION THREAD

**Discussion by Liwoj** [eslint/eslint#15756] [Q&A] [answered]
Is there a simple way how to find current configuration for a given rule?

May sound silly but in today world of shared configs extending different shared configs multiple levels down it is really hard to find out what is the config of some specific rule (basic) **and** from which config it comes from (optional but also useful)

Basic implementation would be in the form of CLI command. Perfect one would give such info from JS API so for example VS Code extension can (after clicking on the reported error) show a menu with "Show documentation for ...." (this exists and is really great) and "Show current settings of ...."

**Comment by mdjermanovic** [comment, marked as answer]
> Basic implementation would be in the form of CLI command.

There is [`--print-config`](https://eslint.org/docs/user-guide/command-line-interface#--print-config) CLI option.

> Perfect one would give such info from JS API so for example VS Code extension can (after clicking on the reported error) show a menu with "Show documentation for ...." (this exists and is really great) and "Show current settings of ...."

Integrations can use [`eslint.calculateConfigForFile()`](https://eslint.org/docs/developer-guide/nodejs-api#-eslintcalculateconfigforfilefilepath)

**Reply by Liwoj** [reply]
> There is [`--print-config`](https://eslint.org/docs/user-guide/command-line-interface#--print-config) CLI option.

Yeah, I was looking at `--print-config` before although something like `--print-rule-config` would be better (I bet it can be done with some terminal magic tho)
 
Thanks!
